pkg/virt-launcher/virtwrap/api/converter.go

```diff
@@ -1432,3 +1441,65 @@ func decoratePciAddressField(addressField string) (*Address, error) {
 	}
 	return decoratedAddrField, nil
 }
+
+func createHostDevForGpu(c *ConverterContext) ([]HostDevice, error) {
+
+	var hostDevices []HostDevice
+	var parseErr error
+	gpuPciAddresses := append([]string{}, c.GpuDevices...)
+	_, isVgpu := os.LookupEnv("IS-VGPU")
+	if isVgpu {
+		hostDevices, parseErr = createVGpuHostDevicesFromPCIAddresses(gpuPciAddresses)
+	} else {
+		hostDevices, parseErr = createHostDevicesFromPCIAddresses(gpuPciAddresses)
+	}
+	if parseErr != nil {
+		log.Log.Reason(parseErr).Error("Unable to parse PCI Device addresses")
+		return nil, parseErr
+	}
+	return hostDevices, nil
+
+}
+
+func createHostDevicesFromPCIAddresses(pcis []string) ([]HostDevice, error) {
+	var hds []HostDevice
+	for _, pciAddr := range pcis {
+		address, err := decoratePciAddressField(pciAddr)
+		if err != nil {
+			return nil, err
+		}
+
+		hostDev := HostDevice{
+			Source: HostDeviceSource{
+				Address: address,
+			},
+			Type:    "pci",
+			Managed: "yes",
+		}
+
+		hds = append(hds, hostDev)
+	}
+
+	return hds, nil
+}
+
+func createVGpuHostDevicesFromPCIAddresses(pcis []string) ([]HostDevice, error) {
+	var hds []HostDevice
+	for _, pciAddr := range pcis {
+		decoratedAddrField := &Address{
+			Uuid: pciAddr,
```

Comment:
Let's call it mdevUUID as this is not a pci address.

Reply:
Fixed